### ISSUE TITLE
[BUG] Same games are listed multiple times

### ISSUE THREAD

**Issue by psychrabi** [Lazap-Development/Lazap#144]
**Describe the bug**
Few games are listed multiple times in the games list

![image](https://github.com/Lazap-Development/Lazap/assets/17595403/399de92e-d5b3-4fea-a415-a7323f39f13e)

**Comment by p0ryae** [comment]
That's probably a bug. Please confirm if deleting `com.lazap.config` and relaunching lazap doesn't fix the issue. If it doesn't, please let me know about where those games are coming from (Steam? Epic Games? etc…). You may find the Enable Launcher Icons toggle useful in the appearances section of the settings.

**Comment by psychrabi** [comment]
Yes games are from steam and epic games and I did deleted folder from app
data for testing the new version

**Comment by p0ryae** [comment]
Please share your `data.json` file so we can investigate this further.

**Comment by LegItMate** [comment]
Insufficient information please do not overwrite the template

**Comment by psychrabi** [comment]
this is different to the #145 is that only after a long time waiting, was i able to get the games list and the game list has duplicates of games as show in the screenshot in #144

I am using Windows 11 Pro build 22635.3420
![image](https://github.com/Lazap-Development/Lazap/assets/17595403/7b250a75-ad8f-4c5a-9677-3d7bb62485ee)

**Comment by LegItMate** [comment]
> Please share your `data.json` file so we can investigate this further.

We can try this now

**Comment by psychrabi** [comment]
https://github.com/Lazap-Development/Lazap/assets/17595403/89c21507-878f-471c-82b6-846202612d18

I tried reinstalling the app after deleting the folder before 
and due to which my data.json is empty
and the game list is not showing at all.
I left the app for some hours and still didnot get the games to list.

**Comment by p0ryae** [comment]
Thanks for the video preview and the description.

Here's the To-Do:
1. Create a personalized build for you so you can check the logs.
2. In the future, we should implument a loading animation (just for end-user) @LegItMate.

**Comment by psychrabi** [comment]
I cloned the repo to my local and ran it and got this error.
perhaps it should shed some light for you.
![image](https://github.com/Lazap-Development/Lazap/assets/17595403/c85e5bcf-75d1-42b0-9c9f-831a26b78976)

I have been following the repo for a while and learnt a log about tauri and rust backend.
Hope to contribute to the repo in future

**Comment by LegItMate** [comment]
> I cloned the repo to my local and ran it and got this error.
> perhaps it should shed some light for you.
> ![image](https://github.com/Lazap-Development/Lazap/assets/17595403/c85e5bcf-75d1-42b0-9c9f-831a26b78976)
> 
> I have been following the repo for a while and learnt a log about tauri and rust backend.
> Hope to contribute to the repo in future

Yes, this is very much helpful

**Comment by LegItMate** [comment]
Do you have any games from uplay launcher?

**Comment by psychrabi** [comment]
Yes i have infact a game in ubisoft - TrackMania
![image](https://github.com/Lazap-Development/Lazap/assets/17595403/c91c258a-852e-4c78-9258-6bc200dbaa62)

**Comment by LegItMate** [comment]
> ![image](https://github.com/Lazap-Development/Lazap/assets/17595403/c85e5bcf-75d1-42b0-9c9f-831a26b78976)

I want you to remove the lines 67-69 in this very same file

**Comment by p0ryae** [comment]
@psychrabi Since you mentioned you compiled lazap from source, please switch to the `nightly` branch and pull the latest changes. 

I have commited 86469c4 which aims to solve the error you're experiencing. 

Let me know if it solves it.

**Comment by psychrabi** [comment]
Yes, i am on nightly branch for testing.

And this has resolve the issue with the games list not loading for long time and loads after few seconds.,
However, i noticed something so i recorded a video and attached here

When the games load, at the first few seconds, there seems to be duplicate of games and then after few seconds the duplicates disappear.

https://github.com/Lazap-Development/Lazap/assets/17595403/b7f8d0f1-f8e6-47ad-8d4e-76918e4337fc

Another thing i noticed is some banners are not showing up and corresponding debug log is show in the terminal
![image](https://github.com/Lazap-Development/Lazap/assets/17595403/260e9a72-d047-4247-a4ab-2a60f1eaeb48)

![image](https://github.com/Lazap-Development/Lazap/assets/17595403/eeca6e97-8a76-40b7-8df2-e05a45e195b9)

**Comment by LegItMate** [comment]
1. It does happen because currently it loads all games and then removes any duplicates. For example, if you bought a GTA game through Epic Games then it will show that GTA game from Epic Games as well as from Rockstar Games, loading the same game twice before removing it.
Some other cases that cause this could be loading of games multiple times in a very short time period, although I expect this to not happen.
And looking at your recording, that's pretty unexpected behaviour.

2. The log means that it did not find a game from the API causing it to use the default banner. This issue will be there for Uplay Launcher's games currently. We will have to look forward to finding a better source for getting banners. Nothing you can do about it really except setting suitable banners manually.

**Comment by p0ryae** [comment]
> 1. It does happen because currently it loads all games and then removes any duplicates. 

Not really the reason for him in this case, because all of his games include this duplicate issue. 

@psychrabi is your `data.json` in good shape? Does it include all of the games? Any duplicates you find in the `data.json`?

**Comment by psychrabi** [comment]
> > 1. It does happen because currently it loads all games and then removes any duplicates.
> 
> Not really sure if this is the reason for him, because almost all of his games include this duplicate issue.
> 
> @psychrabi is your `data.json` in good shape? Does it include all of the games? Any duplicates you find in the `data.json`?

It seems the app also detected my previous installations to different drive.
which i deleted the games and reinstalled to the default paths.

which is why it showed few duplicate games (from 2 locations)

I will try again in a fresh windows install and see.
I have attached my [data.json](https://github.com/Lazap-Development/Lazap/files/15046682/data.json) here

**Comment by psychrabi** [comment]
I did as i mentioned above and deleted all manifest files form older install path and now there is no duplicate games showing up .
![image](https://github.com/Lazap-Development/Lazap/assets/17595403/f7f75f99-3d80-4450-9ca8-ddf2d7a16af6)

**Comment by p0ryae** [comment]
> It seems the app also detected my previous installations to different drive. which i deleted the games and reinstalled to the default paths.

Were they from Steam? Which Launcher(s) specifically if you can recall? Might be helpful to know in order to make the detection better.

> I did as i mentioned above and deleted all manifest files from older install path and now there is no duplicate games showing up .

Please confirm if your problem is resolved. Also, is the loading faster (or improved) compared to the previous time?

**Comment by psychrabi** [comment]
There were from Epic Games
Only the game TrackMania was in both Epic games and Ubisoft with different names
and in Epic Games, if i deleted the game from one directly and reinstalled it on another the it was creating a new manifest file (*.item) in the manifest folder of epic games in programdata\epic

which seems to have caused the duplicate games as Lazap was reading all (*.item) files at first then removing the duplicate names afterwards

**Comment by psychrabi** [comment]
Issue is resolved as of yet.
